src/GraphCoarsenBuilder.hpp

```diff
@@ -148,10 +164,21 @@ class ElementMBuilder : public CoarseMBuilder
     void FillEdgeCdofMarkers(int face_num, const mfem::SparseMatrix& face_Agg,
                              const mfem::SparseMatrix& Agg_cdof_edge);
 
-    std::unique_ptr<mfem::SparseMatrix> GetCoarseM();
+    /// Here returns a null pointer
+    /// @todo change interface so this is optional?
+    std::unique_ptr<mfem::SparseMatrix> GetCoarseM(
+        const mfem::Vector& fineMdiag,
+        const mfem::SparseMatrix& Pedges, const mfem::SparseMatrix& face_cdof);
+
+    bool NeedsCoarseVertexDofs() { return true; }
+
+    /// @todo does DenseMatrix have a move constructor? will this do a deep copy?
```

Comment:
This shouldn't deep copy unless the caller requests a copy, right?
``` c++
const std::vector<DenseMatrix>& CM_el = element_builder.GetElementMatrices(); // No copy
std::vector<DenseMatrix> CM_el_copy = element_builder.GetElementMatrices(); // Deep copy
```
Where the move constructor is never involved, `ElementMBuilder` always keeps its version of the data.